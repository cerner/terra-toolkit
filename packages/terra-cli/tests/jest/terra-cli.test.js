const childProcess = require('child_process');
const path = require('path');
const { promisify } = require('util');

const exec = promisify(childProcess.exec);

describe('terra-cli', () => {
  it('returns the available commands for a non terra-toolkit project', async () => {
    const actualOutput = (await exec('../../../../bin/terra --help', { cwd: path.join(__dirname, 'fixtures', 'normal-project') })).stdout;
    expect(actualOutput).toMatchSnapshot();
  });

  it('returns the available commands for terra-toolkit by looking in the packages directory', async () => {
    const actualOutput = (await exec('../../../../bin/terra --help', { cwd: path.join(__dirname, 'fixtures', 'terra-toolkit') })).stdout;
    expect(actualOutput).toMatchSnapshot();
  });
});
