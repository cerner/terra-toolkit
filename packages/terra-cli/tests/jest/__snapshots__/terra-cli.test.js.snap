// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`terra-cli returns the available commands for a non terra-toolkit project 1`] = `
"Usage: terra <command> [options]

Commands:
  terra command-1   Run command-1
  terra command-2   Run command-2
  terra command-3   Run command-3
  terra command-4   Run command-4
  terra command-11  Run command-11
  terra command-12  Run command-12

Options:
  -h, --help     Show help  [boolean]
  -v, --version  Show version number  [boolean]
"
`;

exports[`terra-cli returns the available commands for terra-toolkit by looking in the packages directory 1`] = `
"Usage: terra <command> [options]

Commands:
  terra command-1  Run command-1
  terra command-2  Run command-2
  terra command-3  Run command-3
  terra command-4  Run command-4

Options:
  -h, --help     Show help  [boolean]
  -v, --version  Show version number  [boolean]
"
`;
